cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -0,0 +1,336 @@
+    out_path = os.path.join(flytekit.current_context().working_directory, "lda.model")
+    lda.save(out_path)
+    return (out_path,)
+
+
+# %%
+# Word Similarities
+# =================
+#
+# We deserialize the model from disk and compute the top 10 similar
+# words to the given word in the corpus (we will use the word `computer` when running
+# the workflow to output similar words). Note that since the model is trained
+# on a small corpus, some of the relations might not be clear.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="600Mi"))
+def word_similarities(model_ser: FlyteFile[MODELSER_NLP], word: str):
+    model = Word2Vec.load(model_ser.path)
+    wv = model.wv
+    logger.info(f"Word vector for {word}:{wv[word]}")
+    logger.info(
+        f"Most similar words in corpus to {word}: {wv.most_similar(word, topn=10)}"
+    )
+
+
+# %%
+# Sentence Similarity
+# ===================
+#
+# We compute Word Mover’s Distance (WMD) using the trained embeddings of words.
+# This enables us to assess the distance between two documents in a meaningful way even when they have
+# no words in common.
+# WMD outputs a large value for two completely unrelated sentences and small value for two closely related
+# sentences.
+# Since we chose two similar sentences for comparison, the word movers distance
+# should be small. You can try altering either ``SENTENCE_A`` or ``SENTENCE_B`` variables to be dissimilar
+# to the other sentence, and check if the value computed is larger.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="600Mi"))
+def word_movers_distance(model_ser: FlyteFile[MODELSER_NLP]) -> float:
+    sentences = [SENTENCE_A, SENTENCE_B]
+    results = []
+    for i in sentences:
+        result = [w for w in utils.tokenize(i) if w not in STOPWORDS]
+        results.append(result)
+    model = Word2Vec.load(model_ser.path)
+    distance = model.wv.wmdistance(*results)
+    logger.info(f"Computing word movers distance for: {SENTENCE_A} and {SENTENCE_B} ")
+    logger.info(f"Word Movers Distance is {distance}")
+    return distance
+
+
+# %%
+# Dimensionality Reduction and Plotting
+# =====================================
+#
+# The word embeddings made by the model can be visualized after reducing the dimensionality to two with t-SNE.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="600Mi"))
+def dimensionality_reduction(model_ser: FlyteFile[MODELSER_NLP]) -> plotdata:
+    model = Word2Vec.load(model_ser.path)
+    num_dimensions = 2
+    vectors = np.asarray(model.wv.vectors)
+    labels = np.asarray(model.wv.index_to_key)
+    tsne = TSNE(n_components=num_dimensions, random_state=0)
+    vectors = tsne.fit_transform(vectors)
+    x_vals = [v[0] for v in vectors]
+    y_vals = [v[1] for v in vectors]
+    return x_vals, y_vals, labels
+
+
+@task(cache_version="1.0", cache=True, limits=Resources(mem="600Mi"))
+def plot_with_matplotlib(x: List[np.float32], y: List[np.float32], labels: np.array):
```

Comment:
np.float32 isn't supported by Flyte yet. Your code should work if you convert numpy to float. Take a look at my suggestions.